Fix a bug that doesn't update MPN field during combination import via file import

buggyzap opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/8/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/8/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR fixes a bug that appear during combinations import via file import procedure.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Upload a file via combinations file import procedure to see the changes and MPN field created/updated.
| Fixed issue or discussion?     | Fixes #35721 

## What fixes this PR

This PR fixes a bug that occurs during combinations import via the official file import procedure.

## How I can reproduce the problem

To reproduce the problem you can simple follow this steps:

1. Go to advanced parameters -> import
2. Select "combinations" on select field
3. Upload a sample combinations file with the MPN field present in the combination row
4. Map the attributes of the files with the Combination attributes
5. Import the file and wait to finish
6. Open the product in the backoffice, you notice that all fields are updated or created but the MPN still remain empty. Everytime.

## Why this happens

After investigating, I've searched in the adminImportController to see what happens with the combinations.
This happens for a simple reason: The MPN field is not present, in the method call that really update the product combination.

## What i've done

I have added the missing MPN field in the method, casting it to string, to update the product.

## Final result

If you retest with this PR you see the MPN field updated.

florine2623 commented:
Could you send your import file please ?

buggyzap commented:
> Could you send your import file please ?

Hi, there are 2 files: products and combinations. You can replicate the issue, importing them, and checking that mpn is not updated in combinations.

[combinations_import_new.csv](https://github.com/PrestaShop/PrestaShop/files/15007839/combinations_import_new.csv)
[products_import(5).csv](https://github.com/PrestaShop/PrestaShop/files/15007840/products_import.5.csv)

florine2623 commented:
@buggyzap ,

I can't import the product file correctly using your file :
<img width="621" alt="Screenshot 2024-04-19 at 14 40 16" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/a07d34aa-6ef8-4a32-b2ac-2207f6d78e47">

I tried with these options :
<img width="665" alt="Screenshot 2024-04-19 at 14 55 11" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/5806e576-4e5c-4044-a3b9-21406c67abc9">

<img width="605" alt="Screenshot 2024-04-19 at 14 42 05" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/b738f958-7a12-4f0c-aaeb-9ff7e640599e">

buggyzap commented:
> @buggyzap ,
> 
> I can't import the product file correctly using your file : <img alt="Screenshot 2024-04-19 at 14 40 16" width="621" src="https://private-user-images.githubusercontent.com/16019289/323964361-a07d34aa-6ef8-4a32-b2ac-2207f6d78e47.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM1NDk5NTEsIm5iZiI6MTcxMzU0OTY1MSwicGF0aCI6Ii8xNjAxOTI4OS8zMjM5NjQzNjEtYTA3ZDM0YWEtNmVmOC00YTMyLWIyYWMtMjIwN2Y2ZDc4ZTQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE5VDE4MDA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyNzNkMjY3NzVlNzY0OGZhMDk5NGIyZWI3NDdlODYwMWZhMmI0ZjM1MmM1ZTYwM2JkNTA0NjgxMzExY2YxNGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NPS1Cu42WxT2sdaysVBUQNdG8PvnkZ6-2HP2ZM8SP4Y">
> 
> I tried with these options : <img alt="Screenshot 2024-04-19 at 14 55 11" width="665" src="https://private-user-images.githubusercontent.com/16019289/323964536-5806e576-4e5c-4044-a3b9-21406c67abc9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM1NDk5NTEsIm5iZiI6MTcxMzU0OTY1MSwicGF0aCI6Ii8xNjAxOTI4OS8zMjM5NjQ1MzYtNTgwNmU1NzYtNGU1Yy00MDQ0LWEzYjktMjE0MDZjNjdhYmM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE5VDE4MDA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2M2JjMDY0YzZkYTZlYmY5OWY1OGMwOTc2N2IxYzA2NDBiNTYwMWUyZTlkYjI1NGM1MDA5YTk3ZGYyZmEzYjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lxwz36h3nfrZBeMS-0QJKqlQ-mAStFm8pAj_i-OQfVI">
> <img alt="Screenshot 2024-04-19 at 14 42 05" width="605" src="https://private-user-images.githubusercontent.com/16019289/323964379-b738f958-7a12-4f0c-aaeb-9ff7e640599e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM1NDk5NTEsIm5iZiI6MTcxMzU0OTY1MSwicGF0aCI6Ii8xNjAxOTI4OS8zMjM5NjQzNzktYjczOGY5NTgtN2ExMi00ZjBjLWFhZWItOWZmN2U2NDA1OTllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE5VDE4MDA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3MjQ2YjFiZWU2MTNkZWFhNGMwYTU0YzlhMWQ0NWIwMWZjMWZhZmQzZTYxODNmODNjN2U1MDE3NjVjNDk1OWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EgCPE5FG-knkxh-yKsl9eYRBV0lNFevGXQQxvg-F2M4">

which map you have used for product attributes? it seems that the field on your side doesn't match

florine2623 commented:
@buggyzap ,

I tried with the 2 files that you shared.

And I also tried with the product example file from PS and I added the MPN values to it.

[products_import (1).csv](https://github.com/user-attachments/files/15789328/products_import.1.csv)

<img width="1435" alt="Screenshot 2024-06-11 at 16 16 00" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/2996f1cf-e369-40fb-a1ba-c50611dfa10f">

florine2623 commented:
friendly reminder @buggyzap

buggyzap commented:
> @buggyzap ,
> 
> I tried with the 2 files that you shared.
> 
> And I also tried with the product example file from PS and I added the MPN values to it.
> 
> [products_import (1).csv](https://github.com/user-attachments/files/15789328/products_import.1.csv)
> <img alt="Screenshot 2024-06-11 at 16 16 00" width="1435" src="https://private-user-images.githubusercontent.com/16019289/338619821-2996f1cf-e369-40fb-a1ba-c50611dfa10f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwODUyMTcsIm5iZiI6MTcyMDA4NDkxNywicGF0aCI6Ii8xNjAxOTI4OS8zMzg2MTk4MjEtMjk5NmYxY2YtZTM2OS00MGZiLWExYmEtYzUwNjExZGZhMTBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA0VDA5MjE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0M2I5NjlmODU5ZWUzMGI2ZDM4YjJiMjAyZjExMjQyYzc4Y2Q3Yjc1NGNjZTk4NWIzZTBkZmE2NDM4ZTg1YTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-vxh6wkLltEsMhOenKU889nT_N-ISW06z7Crvkn7PDw">

Hi this is not related to my file change, I've just added the missing field in file that affect product changes. Other import errors depends on which files are imported.

If you can test a normal product and combination import, successfully, you can just try that the MPN field in the combinations are not updated. With my changes is updated.

If you can't import any entity, there is a big problem on Prestashop, but I don't think that. It's too much effort for me to search files and provide to you the exact configuration as the bugfix doesn't depend on specific data, just the MPN field presence.

Thank you

florine2623 reviewed (commented):
Hello @buggyzap ,

I mapped the MPN field to the UPC field from my import file for testing purposes, I imported the combination file.
<img width="125" alt="Screenshot 2024-07-05 at 14 49 03" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/25ce4820-47ff-4161-a67b-17d4291e053d">

When I check my MPN field in the combination of my product I am able to find them :

https://github.com/PrestaShop/PrestaShop/assets/16019289/70684b55-22b2-43b6-a3b9-529ead2b27cd

The only issue I find is that for one of my field, there's a 0 at the beginning of the MPN number :
I should have `80064075921`.
I have : 
<img width="336" alt="Screenshot 2024-07-05 at 14 53 55" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/1dd693f8-31e1-400c-9f19-46e7d06ca534">

Could you take a look ?
Thanks!

florine2623 commented:
friendly reminder @buggyzap
